### PR TITLE
RavenDB-22983 - clone compressed document item in ForgetAboutDocument [v6.0]

### PR DESCRIPTION
**Pull request by garayx**
### Issue link

* https://issues.hibernatingrhinos.com/issue/RavenDB-22983/AVE-in-SubscriptionConnectionBase-Document.EnsureMetadata-in-5.4.203

* https://github.com/ravendb/ravendb/pull/19557

### Additional description
This pull request includes several changes to the `Raven.Server` project, focusing on improving compressed document handling, which let us remove the calls to `ForgetAbout` method and remove `ForgetAboutEnumerator` class. 
The most important changes include updating the `Document` class to manage document context and storage IDs which let us call ForgetAbout() on document dispose, and refactoring methods to remove calls to `ForgetAbout`.
Also we add flag to force running tests with document compression.

### Document Class Improvements:
* Introduced new fields `_docsContext` and `_storageId` to the `Document` class and provided constructors to initialize them. (`src/Raven.Server/Documents/Document.cs`) [[1]](diffhunk://#diff-711a8a675e2d394be3d6ef5af3478cd944f0e2fa4d237f81a90f522d7fdd8453R39-R40) [[2]](diffhunk://#diff-711a8a675e2d394be3d6ef5af3478cd944f0e2fa4d237f81a90f522d7fdd8453R53-R63)
* Updated the `Dispose` method in the `Document` class to call `ForgetAbout` on the transaction using `_storageId`. (`src/Raven.Server/Documents/Document.cs`)

### License Management Enhancements:
* Added `IgnoreCompressionLicenseLimit` flag to the `LicenseManager` class and updated `AssertCanUseDocumentsCompression` to respect this flag. (`src/Raven.Server/Commercial/LicenseManager.cs`) [[1]](diffhunk://#diff-8ab40658524634ad957639363f4eef427b5b62d65ce657afcb647e895ebed48aR89) [[2]](diffhunk://#diff-8ab40658524634ad957639363f4eef427b5b62d65ce657afcb647e895ebed48aR1419-R1421)

### Refactoring `ForgetAbout` Method:
* Removed calls to `ForgetAbout` in various methods and classes, including `CollectionRunner`, `ExpirationStorage`, `StreamingHandler`, and several others. (`src/Raven.Server/Documents/CollectionRunner.cs`, `src/Raven.Server/Documents/Expiration/ExpirationStorage.cs`, `src/Raven.Server/Documents/Handlers/Streaming/StreamingHandler.cs`, etc.) [[1]](diffhunk://#diff-529d5a8892c15bf919213c4d8b90edc3240353cfd78b2b43813cf71fe9f5efaeL127-L128) [[2]](diffhunk://#diff-be34884e1bd779584e34af0721515c99e9c5358fd4a3663c19f8f72c1de29b9dL159) [[3]](diffhunk://#diff-e7da32d349abb8cf09b585badcd3f96002bbeed8619b755ff879acc3fe45ba3fL53-R53) etc.)

### Context Management in Documents:
* Updated methods to use `DocumentsOperationContext` instead of `JsonOperationContext` and refactored document parsing methods to initialize documents with context and storage IDs. (`src/Raven.Server/Documents/DocumentsStorage.cs`) [[1]](diffhunk://#diff-624ee6c1d1af1fe87f3d169216fa6ea981b9835799020d41b1625cb7bb450ec5L1567-R1595) [[2]](diffhunk://#diff-624ee6c1d1af1fe87f3d169216fa6ea981b9835799020d41b1625cb7bb450ec5L1619-R1625)

### Miscellaneous Changes:
* Removed unnecessary nested classes and methods related to `ForgetAbout` in `DocumentReplicationItem` and other classes. (`src/Raven.Server/Documents/Replication/ReplicationItems/DocumentReplicationItem.cs`) [[1]](diffhunk://#diff-4bbb042ee7a808d48f5e1a1154dad9a91200b2f4271518929daee4989af06531L44-L45) [[2]](diffhunk://#diff-4bbb042ee7a808d48f5e1a1154dad9a91200b2f4271518929daee4989af06531L213-L229)
### Type of change

- [ ] Bug fix
- [x] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

**Comment by garayx** [comment]
test this please

File: test/Tests.Infrastructure/RavenTestBase.cs [hunk 1]
```diff
@@ -784,6 +834,7 @@ public class Options
             private X509Certificate2 _clientCertificate;
             private X509Certificate2 _adminCertificate;
             private bool _createDatabase;
+            private bool _ignoreDocumentCompression;
```
Comment (arekpalinski) [inline test/Tests.Infrastructure/RavenTestBase.cs]: Please check the warning here

Comment (garayx) [reply test/Tests.Infrastructure/RavenTestBase.cs]: done

**Review by arekpalinski — APPROVED** [review]

**Comment by ppekrol** [comment]
Can you rebase?

**Comment by garayx** [comment]
> Can you rebase?

done

**Pull request merged**